src/main/scala/com/cognite/sdk/scala/v1/Client.scala

```diff
@@ -216,13 +199,49 @@ object GenericClient {
     deriveDecoder[ProjectAuthentication]
   @SuppressWarnings(Array("org.wartremover.warts.JavaSerializable"))
   implicit val projectDecoder: Decoder[Project] = deriveDecoder[Project]
+
+  val defaultBaseUrl = Option(System.getenv("COGNITE_BASE_URL"))
+    .getOrElse("https://api.cognitedata.com")
+
+  def parseBaseUrlOrThrow(baseUrl: String): Uri =
+    try {
+      uri"$baseUrl"
+    } catch {
+      case NonFatal(_) =>
+        throw new IllegalArgumentException(
+          s"Unable to parse baseUrl = ${baseUrl} as a valid URI."
+        )
+    }
+
+  def forAuth[F[_]: Monad, S](
+      applicationName: String,
+      auth: Auth,
+      baseUrl: String = defaultBaseUrl
+  )(implicit sttpBackend: SttpBackend[F, S]): F[GenericClient[F, S]] = {
+    val login = new Login[F](
+      RequestSession(applicationName, parseBaseUrlOrThrow(baseUrl), sttpBackend, auth)
+    )
+    for {
+      status <- login.status()
+      projectName = status.project
+    } yield
+      if (projectName.trim.isEmpty) {
+        throw InvalidAuthentication()
```

Comment:
wondering if this is something that should happen already in `login.status()`. Then the user of this function does not have to take care of a call going wrong. Respectively if the returntype of `status` changes to something like `OptionT[F, LoginStatus]` it would be obvious through the type system that something went off track

Reply:
For now at least, we're keeping the return types simple to keep the door at least somewhat open for Java compatibility.

Since we have cross-project service accounts nowadays I also think such a `login.status()` return value doesn't necessarily mean that anything is wrong, but it would be useless here since the purpose is to retrieve the project name to use in URLs against other resource types.